autolab/Autolab
# [Breaking] Use ActiveStorage for attachments, update attachments tests

## Description
- Use ActiveStorage to store and serve attachment files
- Update existing tests

**This is a breaking change, and old attachments will have to be re-uploaded.**

## Motivation and Context
Currently, all attachments are stored in a single top-level directory, where collisions are resolved by appending numbers to the end of the filename. This PR switches to ActiveStorage, which handles the file storage logic for us.

Using ActiveStorage also makes it easy to switch to other services such as S3 in the future.

Resolves #617 (albeit with a different approach)

More info about ActiveStorage: https://guides.rubyonrails.org/active_storage_overview.html

Note: Undoes the changes made in #1490 (in consultation with the author) in preference for a better solution to the XSS issue. 
In that PR, the mime_type was set to `application/octet-stream` to force downloads. A better solution is to remove `disposition: inline` since the default is `disposition: attachment`, which forces downloads.

More details here: https://huntr.dev/bounties/90701766-bfed-409e-b3dd-6ff884373968/

## How Has This Been Tested?
ActiveStorage

- Either clear attachments folder or remember what it looks like (or delete your attachments folder)
- Create an attachment
  - Observe that `/attachments` remains unchanged
  - Observe that attachment is stored in `/storage` (might be several directories down, with a randomly generated file name)
- Download the attachment
- Edit and download the attachment
- Delete attachment, observe that file is deleted from `/storage`
- If you want, you can test that the released / unreleased options work, but this PR doesn't touch that logic

XSS fix
- Upload example file from huntr.dev report: ![something](https://user-images.githubusercontent.com/9074856/219969212-bbfbddb9-1c7f-4b2f-814a-c6c08a2e4f45.svg)
- When attachment is clicked, ensure file is downloaded rather than displayed

Tests
- Run `rake spec SPEC=./spec/controllers/attachments_controller_spec.rb` and ensure they pass
- Also run the full test suite

## Types of changes

## Checklist:
